stdlib/src/random/random.mojo

```diff
@@ -201,3 +201,19 @@ fn randn[
     for i in range(size):
         ptr[i] = randn_float64(mean, variance).cast[type]()
     return
+
+
+fn shuffle[T: CollectionElement, //](inout list: List[T]):
+    """Shuffles the elements of the list randomly.
+
+    Performs an in-place Fisher-Yates shuffle on the provided list.
+
+    Args:
+        list: The list to modify.
+
+    Parameters:
+        T: The type of element in the List.
+    """
+    for i in reversed(range(len(list))):
+        var j = int(random_ui64(0, i))
+        swap(list.unsafe_get(i), list.unsafe_get(j))
```

Comment:
**Question** Should we use the new `List.swap_elements` function instead for exclusivity checking purposes?

Reply:
Sounds good, missed the introduction of this.